### PR TITLE
Port scheduler latency improvements to 3.6.

### PR DESCRIPTION
**Pull request by neunhoef**

### CODE REVIEW

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/7706/

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/7706/

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-ANY-matrix/4/

**Comment by KVS85** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-ANY-matrix/5/

**Comment by KVS85** [comment]
Tests blue.

File: arangod/Scheduler/SupervisedScheduler.h [hunk 1]
```diff
@@ -130,13 +135,21 @@ class SupervisedScheduler final : public Scheduler {
...
     bool start();
   };
-  size_t _maxNumWorker;
-  size_t _numIdleWorker;
+  size_t const _minNumWorker;
+  size_t const _maxNumWorker;
   std::list<std::shared_ptr<WorkerState>> _workerStates;
   std::list<std::shared_ptr<WorkerState>> _abandonedWorkerStates;
-
+  std::atomic<uint64_t> _nrWorking;   // Number of threads actually working
```
Comment (joerg84) [inline arangod/Scheduler/SupervisedScheduler.h]: 1. Would it make sense to export these as metrics?

**Pull request merged**